### PR TITLE
Login controller and singleton implementation

### PR DESCRIPTION
**Pull request by murtiturti**

### CODE REVIEW

File: src/main/java/useCases/LogIn.java [hunk 1]
```diff
@@ -13,14 +13,20 @@
...
+    private final String username;
+    private final String password;
+
+    public LogIn(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
 
     /**
-     * @param username: username of the User object to deserialize
```
Comment (jaimans) [inline src/main/java/useCases/LogIn.java:18]: add parameters for each file including login

**Review by jaimans — COMMENTED** [review]
Clean architecture violation fixed - Create a controller to bridge the gap in the dependency. 

- Login call to use case is deleted

File: src/main/java/CommandLineInterface.java [hunk 2]
```diff
@@ -55,7 +56,7 @@ else if (entry == 1)
                 // Do Login
                 try {
                     // Try login
-                    User userLoggedIn = LogIn.login(username, password);
+                    User userLoggedIn = (User) LoginController.getInstance().login(username, password);
```
[the remark this answers is not in the document]
Comment (shaunamia) [reply src/main/java/CommandLineInterface.java:59]: looks good!

File: src/main/java/useCases/LogIn.java (hunk 1 above)
Comment (shaunamia) [reply src/main/java/useCases/LogIn.java:18]: doesn't cause any conflicts elsewhere, good!

**Comment by shaunamia** [comment]
Code can be merged, no conflicts elsewhere, and changes made to improve clean architecture make sense.

**Pull request merged**